### ISSUE TITLE
Vyhledávání na Ulož.to

### ISSUE THREAD

**Issue by exander77** [setnicka/ulozto_downloader#12]
Mám napsané vyhledávání na Ulož.to v NodeJS, proto jsem přidal podporu trackovaných linků, aby se linky z mého vyhledávače daly rovnou používat v ulozto_downloader. Získat netrackované linky znamená pro každý link z vyhledávání provést jeden dotaz na Ulož.to.

Trackované linky vypadají takto:
`https://uloz.to/download-tracking/50bb81ef93310e4266e9b2b85199c595a9b8a576a8f3e5ad4c8f3b8dc377107eada50b3d873de2b6e8fec7a9bb2306f6`

Budu to dávat brzy asi na Github a mohly by ty projekty spolupracovat.

**Comment by ZDVokoun** [comment]
Protože zatím @exander77 svůj program na vyhledávání nezveřejnil, rozhodl jsem se pokusit ho napsat sám. Napsal jsem ho v Pythonu. Pokud byste měli zájem to přidat, zde je kód: [ZDVokoun/ulozto-search](https://github.com/ZDVokoun/ulozto-search). Doufám, že to pomůže.
PS: omlouvám se za svůj nečistý kód, programuji jen krátce

**Comment by exander77** [comment]
@ZDVokoun It was published quite a long time ago, but I hadn't maintained it: https://github.com/exander77/foolozto
And I got no response from authors of this repo.

**Comment by ZDVokoun** [comment]
> @ZDVokoun It was published quite a long time ago, but I hadn't maintained it: https://github.com/exander77/foolozto And I got no response from the authors of this repo.

Oh. I'm sorry, I couldn't find it under that name. That sucks that they didn't respond to you.

**Comment by ZDVokoun** [comment]
For others, if you want to download all files returned after querying uloz.to, install [ulozto-search](https://github.com/ZDVokoun/ulozto-search), open a terminal and enter:
```sh
ulozto-search "your query" | while read in; do ulozto-downloader --auto-captcha $(echo "$in" | cut -d '|' -f2); done
```
I hope this helps.

**Comment by stinovlas** [comment]
I think that `ulozto-search` might deserve mention in README (we could add `Related projects` section).

On a related note, I'd like to add a switch that would allow `ulozto-downloader` to download list of files instead of just one file at a time. You could specify maximum number of simultaneous downloads and when some parts of one file are already downloaded, `ulozto-downloader` would use those free threads to start download of another file. This might speed up serialized downloads – now we have to wait for all downloads of one file to finish before we start another one (unless we run multiple `ulozto-downladers` in parallel, but that also not what I want). I might get to it during the Christmas, but no promises.

**Comment by ZDVokoun** [comment]
That would be great, but [the maintainer](https://github.com/setnicka) of this repo wasn't active on Github for almost two months.

**Comment by setnicka** [comment]
Yes, unfortunately I did not have a much free time and energy for this hobby project in last few month. Pull requests are certainly welcomed and I will tr to review them as quick as I manage.
